src/App.tsx

```diff
@@ -15,37 +16,231 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App: React.FC = () => {
+enum SortType {
+  Default = '',
+  Length = 'LENGTH',
+  Alphabet = 'ALPHABET',
+}
+
+interface FilterParams {
+  sortField: SortType | '';
+  isReversed: boolean;
+}
+
+function getPreperedGoods({ sortField, isReversed }: FilterParams): string[] {
+  const preperedGoods = [...goodsFromServer];
+
+  if (sortField) {
+    preperedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SortType.Alphabet:
+          return good1.localeCompare(good2);
+
+        case SortType.Length:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReversed) {
+    preperedGoods.reverse();
+  }
+
+  return preperedGoods;
+}
+
+export const App = () => {
+  const [sortField, setsortField] = useState<SortType>(SortType.Default);
+  const [isReversed, setIsReversed] = useState(false);
+
+  const visibleGoods = getPreperedGoods({ sortField, isReversed });
+  const sortedOrReversedList = sortField || isReversed;
```

Comment:
```suggestion
  const isResetActive= sortField || isReversed;
```
Let's simplify this variable